core/shared/src/main/scala/org/http4s/Uri.scala

```diff
@@ -657,14 +657,25 @@ object Uri extends UriPlatform {
       * the underlying IP protocol version of the given address, either IPv4
       * or IPv6.
       */
-    def fromIpAddress(value: ip4s.IpAddress): Host =
+    @inline def fromIpAddress(value: ip4s.IpAddress): Host =
```

Comment:
Are we sure that `@inline` annotation will never generate a warning at the callsite?  Not opposed to it, but I've never been confident enough its ramifications to use it myself.  The rest looks good.

Reply:
What kind of warning?

FWIW I'm leery about `@inline` myself, esp. setting precedent to decide whether we should `@inline` a method or not. Happy to leave it to the JIT.

Reply:
Quoth [the docs](https://www.scala-lang.org/api/2.13.4/scala/inline.html):

> When inlining is enabled, the inliner will always try to inline methods or callsites annotated @inline (under the condition that inlining from the defining class is allowed, see -opt-inline-from:help). If inlining is not possible, for example because the method is not final, an optimizer warning will be issued. See -opt-warnings:help for details.

They give one example.  Are there more?  I live a life of fear.

Reply:
Can't prove that users won't face any warnings. So let's remove it. In my defense, I can say that I never faced inline-based warnings but faced using `@inline` for patmat-based methods.

Reply:
Yeah, sorry to bring paranoia instead of evidence.  I got excited about `@inline` several Scalae ago and was warned it was for special occasions.  I've just started trusting the defaults until I have a smoking benchmark.